### ISSUE TITLE
aws_db_event_subscription deletion is failing with `Error deleting RDS Event Subscription`

### ISSUE THREAD

**Issue by keshav** [terraform-providers/terraform-provider-aws#9243]
#### Issue
While trying to delete an existing aws_db_event_subscription, terraform apply is failing with following error. 
```

aws_db_event_subscription.some_name: Still destroying... (ID: some_id, 30s elapsed)
--
  |  
  | Error: Error applying plan:
  |  
  | 1 error occurred:
  | * aws_db_event_subscription.some_name (destroy): 1 error occurred:
  | * aws_db_event_subscription.some_name: Error deleting RDS Event Subscription some_id : Unable to find RDS Event Subscription: []*rds.EventSubscription(nil)
  | 
```
Terraform plan is showing deletion of RDS subscription event correctly but during the apply phase, it's throwing the above error. Once the terraform apply is failed, all the events are deleted in the  actual AWS account, but terraform plan still has the RDS event entries in the state file. As a result, Terraform plan is failing for all the subsequent builds. Below is the plan failure
```
Error: Error refreshing state: 1 error occurred:
--
  | * aws_db_event_subscription.some_name: aws_db_event_subscription.some_name: Error retrieving RDS Event Subscription some_id : Unable to find RDS Event Subscription: []*rds.EventSubscription(nil)
```

### Steps to reproduce
* Create a AWS RDS subscription event 
* Try to delete the event and apply the change

### Affected Versions
* Terraform Version: 0.11.14
* Terraform Aws Provider version: 2.17.0

**Comment by bflad** [comment]
Bug fix submitted: https://github.com/terraform-providers/terraform-provider-aws/pull/9274

**Comment by bflad** [comment]
The fix for this has been merged and will be released with version 2.19.0 of the Terraform AWS Provider, likely in two days. 👍

**Comment by bflad** [comment]
This has been released in [version 2.19.0 of the Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#2190-july-11-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

For further feature requests or bug reports with this functionality, please create a [new GitHub issue](https://github.com/terraform-providers/terraform-provider-aws/issues/new/choose) following the template for triage. Thanks!

**Comment by dkowalcz** [comment]
I just ran into the same issue using version2.19.0 - the destroy removed the event from AWS but left the entry in the statefile: 

* provider.aws: version = "~> 2.19"

Terraform has been successfully initialized!

aws_db_event_subscription.EventSubscriptionAlarmInstance: Refreshing state... (ID: isg-mysql-demo4-eventsubscriptionalarminstance)

Error: Error refreshing state: 1 error(s) occurred:

* aws_db_event_subscription.EventSubscriptionAlarmInstance: 1 error(s) occurred:

* aws_db_event_subscription.EventSubscriptionAlarmInstance: aws_db_event_subscription.EventSubscriptionAlarmInstance: error retrieving RDS Event Subscription (isg-mysql-demo4-eventsubscriptionalarminstance): SubscriptionNotFound: Event Subscription isg-mysql-demo4-eventsubscriptionalarminstance not found.
	status code: 404, request id: b8ec0a3f-e84e-4eef-9e6b-adf15bb88d69


Statefile: 
                    "primary": {
                        "id": "isg-mysql-demo4-eventsubscriptionalarminstance",
                        "attributes": {
                            "arn": "arn:aws:rds:us-east-1:############:es:isg-mysql-demo4-eventsubscriptionalarminstance",
                            "customer_aws_id": "############",
                            "enabled": "true",
                            "event_categories.#": "12",
                            "event_categories.1068999359": "availability",
                            "event_categories.1072746924": "backup",
                            "event_categories.1866582008": "restoration",
                            "event_categories.2890955135": "configuration change",
                            "event_categories.3037192942": "low storage",
                            "event_categories.3102116255": "failover",
                            "event_categories.3209983690": "notification",
                            "event_categories.3430174804": "read replica",
                            "event_categories.4081292233": "recovery",
                            "event_categories.563807169": "failure",
                            "event_categories.769513765": "deletion",
                            "event_categories.797243625": "maintenance",
                            "id": "isg-mysql-demo4-eventsubscriptionalarminstance",
                            "name": "isg-mysql-demo4-eventsubscriptionalarminstance",
                             ................
